File: mobile/lib/features/pagos/data/models/pago_model.dart
class PagoModel {
  final int id;
  final int idIncidente;
  final double montoTotal;
  final double montoComision;
  final double montoTaller;
  final String metodoPago;
  final String? referenciaExterna;
  final String estadoPago;
  final bool comisionPagada;
  final DateTime creadoEn;
  final DateTime? pagadoEn;

  const PagoModel({
    required this.id,
    required this.idIncidente,
    required this.montoTotal,
    required this.montoComision,
    required this.montoTaller,
    required this.metodoPago,
    this.referenciaExterna,
    required this.estadoPago,
    required this.comisionPagada,
    required this.creadoEn,
    this.pagadoEn,
  });

  factory PagoModel.fromJson(Map<String, dynamic> json) => PagoModel(
        id: json['id'],
        idIncidente: json['id_incidente'],
        montoTotal: (json['monto_total'] as num).toDouble(),
        montoComision: (json['monto_comision'] as num).toDouble(),
        montoTaller: (json['monto_taller'] as num).toDouble(),
        metodoPago: json['metodo_pago'] ?? 'tarjeta',
        referenciaExterna: json['referencia_externa'],
        estadoPago: json['estado_pago'] ?? 'pendiente',
        comisionPagada: json['comision_pagada'] ?? false,
        creadoEn: DateTime.parse(json['creado_en']),
        pagadoEn: json['pagado_en'] != null
            ? DateTime.parse(json['pagado_en'])
            : null,
      );
}

File: Backend/app/models/pago_model.py
from sqlalchemy import (
    Column, Integer, Numeric, String, Boolean, TIMESTAMP, ForeignKey, func, Text
)
from sqlalchemy.orm import relationship
from app.core.db import Base


class Pago(Base):
    __tablename__ = "pagos"

    id                = Column(Integer, primary_key=True, index=True)
    id_incidente      = Column(Integer, ForeignKey("incidentes.id"), unique=True, nullable=False)
    id_usuario        = Column(Integer, ForeignKey("usuario.id"), nullable=False)
    id_taller         = Column(Integer, ForeignKey("talleres.id"), nullable=False)
    monto_total       = Column(Numeric(10, 2), nullable=False)
    monto_comision    = Column(Numeric(10, 2), nullable=False)  # 10%
    monto_taller      = Column(Numeric(10, 2), nullable=False)  # 90%
    metodo_pago       = Column(String(30), nullable=False, default="tarjeta")
    referencia_externa = Column(Text, nullable=True)   # payment_intent_id de Stripe
    estado_pago       = Column(String(20), nullable=False, default="pendiente")
    # pendiente | completado | fallido | reembolsado
    comision_pagada   = Column(Boolean, nullable=False, default=False)
    creado_en         = Column(TIMESTAMP(timezone=True), nullable=False, server_default=func.now())
    pagado_en         = Column(TIMESTAMP(timezone=True), nullable=True)

    # Relaciones
    incidente = relationship("Incidente", back_populates="pago")
    usuario   = relationship("Usuario")
    taller    = relationship("Taller")